build script

0xSTAR opened the pull request:
for the potential of redistributables in the Releases tab :)
also adds a requirements.txt which lists dependencies for the build script.
This allows people who want to "build" from source themselves to easily install dependencies like so:
        `(INTERPRETER) -m pip install -r requirements.txt`

(INTERPRETER) is whatever you have your Python binary as...

I've commented it fairly.

0xSTAR committed:
add a build script 

for the potential of redistributables in the Releases tab :)

ArthurHeitmann commented:
Hi, this looks good. I saw that the `--decompileAll` option is a separate config option in the script. This didn't quite feel right, so I made it now redundant. `__init__.py` will now automatically detect if it's a folder or a file.

Could you please also do the following things and I'll merge it.
- As I said the batch option is no longer needed
- Admin should be `False` by default. I didn't have any problems with it.
- Add the the `dist` folder to the `.gitignore`
- If you want, add a small section in the readme with an examplanation
- A question: How are the precompiled mruby binaries added to the generated file? Does the tool do it automatically?

I'll have yet to see if I add it to the releases tab, since that's an additional step after each push and I'm a bit lazy. But otherwise this is really solid.

0xSTAR committed:
Merge branch 'ArthurHeitmann:master' into master

0xSTAR committed:
fix build script accordingly + gitignore + README update

fixes the build script according to specified in your comment on the initial pull request.
also adds necessities to the .gitignore, and updated the README. 
Changes to the README also make the original text in the README more readable (lol).

I recommend doing versioning by the day the binary was built, rather than semantics (X.X.X). 
Kinda like how Arch does their rolling releases, and the ISO is like YY.MM.DD (year, month, day)

0xSTAR committed:
update gitignore

0xSTAR commented:
I have updated according to what you have listed. @ArthurHeitmann (sorry for the direct mention)
And also about the question

_A question: How are the precompiled mruby binaries added to the generated file? Does the tool do it automatically?_

If you mean that the precompiled MRuby binaries are bundled with the frozen binary, then no.
The user will need to feed it the directory just like how they would the script, except its just one file they need, rather than to get the whole repository. Also would allow for versioning (preferably by DATE. like how Arch Linux does YY.MM.DD (year, month, date)). In the case of potential instability in the repo, the Release would be superior. This also allows the decompiler to be much more easily redistributed.

ArthurHeitmann commented:
Sorry that it took some time. Looks all good.
